superset/connectors/sqla/models.py

```diff
@@ -517,27 +517,6 @@ def data(self) -> Dict[str, Any]:
 )
 
 
-def _process_sql_expression(
-    expression: Optional[str],
-    database_id: int,
-    schema: str,
-    template_processor: Optional[BaseTemplateProcessor] = None,
-) -> Optional[str]:
-    if template_processor and expression:
-        expression = template_processor.process_template(expression)
-    if expression:
-        try:
-            expression = validate_adhoc_subquery(
-                expression,
-                database_id,
-                schema,
-            )
-            expression = sanitize_clause(expression)
-        except (QueryClauseValidationException, SupersetSecurityException) as ex:
-            raise QueryObjectValidationError(ex.message) from ex
-    return expression
-
-
```

Comment:
This method was almost identical to the base class, the only difference is that is handles `SupersetSecurityException`. I updated the base class method and removed this implementation.